Add explicit timeout to flush API (#11008)
rerun-io/rerun: docs/content/reference/migration/migration-0-25.md

Use `--web-viewer` instead.

## Flush takes an optional timeout, and returns errors
When flushing a recording stream you can now give it a maximum time for how long it should block.

> The flush will block until either it completes, fails, or the timeout is reached.

Reviewer: ```suggestion
The flush will block until either it completes, fails (e.g. because of connection loss), or the timeout is reached.
```

> Previously this could only be configured for gRPC sinks, and it was configured once then first connecting.

Reviewer: ```suggestion
Previously this could only be configured for gRPC sinks, and it was configured once upon setting up the connection.
```
